Marlin/ultralcd.cpp

```diff
@@ -3527,59 +3522,64 @@ void lcd_update() {
+              encoderPosition -= ENCODER_STEPS_PER_MENU_ITEM;
+            }
+            else if (buttons_reprapworld_keypad&EN_REPRAPWORLD_KEYPAD_UP) {
+              encoderPosition += ENCODER_STEPS_PER_MENU_ITEM;
+            }
+            else if (buttons_reprapworld_keypad & EN_REPRAPWORLD_KEYPAD_LEFT) {
+              menu_action_back();
+            }
+            else if (buttons_reprapworld_keypad&EN_REPRAPWORLD_KEYPAD_RIGHT) {
+              // enqueue_and_echo_commands_P(PSTR("M0 Pause"));
+              lcd_return_to_status();
+            }
+          }
+          else {
+            int32_t step = 1;
+            if (adc_steps > 10) {
+              step = 10;
+            }
+            if (adc_steps > 19) {
+              step = 100;
+            }
+            if (buttons_reprapworld_keypad&EN_REPRAPWORLD_KEYPAD_DOWN) {
+              encoderPosition += ENCODER_PULSES_PER_STEP  * step;
+            }
+            else if (buttons_reprapworld_keypad&EN_REPRAPWORLD_KEYPAD_UP) {
+              encoderPosition -= ENCODER_PULSES_PER_STEP  * step;
+            }
+            else if (buttons_reprapworld_keypad&EN_REPRAPWORLD_KEYPAD_RIGHT) {
+              encoderPosition = 0;
+            }
+          }
+          #ifdef ADC_KEYPAD_DEBUG
+            SERIAL_PROTOCOLPGM("buttons_reprapworld_keypad = ");
+            SERIAL_PROTOCOLLN((unsigned long)buttons_reprapworld_keypad);
+            SERIAL_PROTOCOLPGM("encoderPosition = ");
+            SERIAL_PROTOCOLLN((unsigned long)encoderPosition);
+          #endif
         }
-        else
-        {
-          int32_t step = 1;
-          if (adc_steps > 10)
-            step = 10;
-          if (adc_steps > 19)
-            step = 100;
-          if (buttons_reprapworld_keypad&EN_REPRAPWORLD_KEYPAD_DOWN)
-            encoderPosition += ENCODER_PULSES_PER_STEP  * step;
-          else if (buttons_reprapworld_keypad&EN_REPRAPWORLD_KEYPAD_UP)
-            encoderPosition -= ENCODER_PULSES_PER_STEP  * step;
-          else if (buttons_reprapworld_keypad&EN_REPRAPWORLD_KEYPAD_RIGHT)
-            encoderPosition = 0;
+        else if(!thermalManager.current_ADCKey_raw) {
+          // reset stepping acceleration
+          adc_steps = 0;
         }
-      #ifdef ADC_KEYPAD_DEBUG
-        SERIAL_PROTOCOLPGM("buttons_reprapworld_keypad = ");
-        SERIAL_PROTOCOLLN((unsigned long)buttons_reprapworld_keypad);
-        SERIAL_PROTOCOLPGM("encoderPosition = ");
-        SERIAL_PROTOCOLLN((unsigned long)encoderPosition);
-      #endif
-      }
-      else if(!thermalManager.current_ADCKey_raw)
-      {
-        // reset stepping acceleration
-        adc_steps = 0;
-      }
 
-    #elif ENABLED(REPRAPWORLD_KEYPAD)
-      handle_reprapworld_keypad();
-	  #endif
+      #elif ENABLED(REPRAPWORLD_KEYPAD)
+        handle_reprapworld_keypad();
+      #endif
 
       bool encoderPastThreshold = (abs(encoderDiff) >= ENCODER_PULSES_PER_STEP);
```

Comment:
i planned to create a "handle_adc_keypad()" function when going to create a marlin PR from the code.